packages/components/date-picker/src/date-picker-com/panel-date-range.vue

```diff
@@ -681,6 +680,16 @@ export default defineComponent({
       clearable,
     } = pickerBase.props
 
+    const shortcuts = ref([])
+    shortcuts.value = pickerBase.props
```

Comment:
This is a little confusing, when you have `shortcut` props injected into `pickerBase.props` why you assign `pickerBase.props` to `shortcuts.value`?

Here we have
```ts
    const shortcuts = ref([])
    shortcuts.value = pickerBase.props
```
is essentially equivalent to
```ts
    const shortcuts = ref(pickerBase.props)
```
Also the statement here **IS WRONG**

it should be either

```ts
    const shortcuts = ref([])
    shortcuts.value = pickerBase.props.shortcuts
```

or

```ts
    const shortcuts = ref(pickerBase.props.shortcuts || [])
```

with the second way you are free to remove `immediate` from there.

Reply:
If you want to keep the `immediate` part, you will only need https://github.com/element-plus/element-plus/pull/3274/files#diff-c6cfc3c5d64d776c7962e059460dc01995f9f692faab74c6d2e3175eed0d80d6R683